libbeat/docs/shared-env-vars.asciidoc

```diff
@@ -49,3 +65,38 @@ and what each configuration looks like after replacement:
 |`name: ${NAME:beats}` |no setting            |`name: beats`
 |`name: ${NAME:beats}` |`export NAME=elastic` |`name: elastic`
 |==================================
```

Comment:
@urso I'm looking at this in 6.0.0-alpha2, and the error message with the custom text isn't what you would expect.  When `name: ${NAME:?You need to set the NAME environment variable}` is set in the config and the NAME env var can't be resolved, the Beat returns the error `You need to set the NAME environment variable accessing 'name' (source:'filebeat.yml')`. (Whatever you specify for custom text gets prepended to `accessing 'name' (source:'filebeat.yml')`.  I'll push the changes I made, but I'm not sure we should document the error syntax since it results in weird messages.

Also, since the alternate value operator is an advanced use case that almost no one will use in practice AND it's tricky to use, I don't think we should document it. We might cause more problems than we solve by documenting it. WDYT?